scipy/stats/tests/test_survival.py

```diff
@@ -49,3 +81,68 @@ def test_nonunique(self):
         assert_equal(res.x, ref_x)
         assert_equal(res.cdf, ref_cdf)
         assert_equal(res.sf, ref_sf)
+
+    # ref. [1] page 91
+    t1 = [37, 43, 47, 56, 60, 62, 71, 77, 80, 81]  # times
+    d1 = [0, 0, 1, 1, 0, 0, 0, 1, 1, 1]  # 1 means deaths (not censored)
+    r1 = [1, 1, 0.875, 0.75, 0.75, 0.75, 0.75, 0.5, 0.25, 0]  # reference SF
+
+    # https://sphweb.bumc.bu.edu/otlt/mph-modules/bs/bs704_survival/BS704_Survival5.html  # noqa
+    t2 = [8, 12, 26, 14, 21, 27, 8, 32, 20, 40]
+    d2 = [1, 1, 1, 1, 1, 1, 0, 0, 0, 0]
+    r2 = [0.9, 0.788, 0.675, 0.675, 0.54, 0.405, 0.27, 0.27, 0.27]
+    t3 = [33, 28, 41, 48, 48, 25, 37, 48, 25, 43]
+    d3 = [1, 1, 1, 0, 0, 0, 0, 0, 0, 0]
+    r3 = [1, 0.875, 0.75, 0.75, 0.6, 0.6, 0.6]
+
+    # https://sphweb.bumc.bu.edu/otlt/mph-modules/bs/bs704_survival/bs704_survival4.html  # noqa
+    t4 = [24, 3, 11, 19, 24, 13, 14, 2, 18, 17,
+          24, 21, 12, 1, 10, 23, 6, 5, 9, 17]
+    d4 = [0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 1, 0, 1]
+    r4 = [0.95, 0.95, 0.897, 0.844, 0.844, 0.844, 0.844, 0.844, 0.844,
+          0.844, 0.76, 0.676, 0.676, 0.676, 0.676, 0.507, 0.507]
+
+    # https://www.real-statistics.com/survival-analysis/kaplan-meier-procedure/confidence-interval-for-the-survival-function/  # noqa
+    t5 = [3, 5, 8, 10, 5, 5, 8, 12, 15, 14, 2, 11, 10, 9, 12, 5, 8, 11]
+    d5 = [1, 1, 1, 0, 1, 0, 1, 1, 0, 0, 1, 1, 0, 0, 1, 1, 1, 1]
+    r5 = [0.944, 0.889, 0.722, 0.542, 0.542, 0.542, 0.361, 0.181, 0.181, 0.181]
+
+    @pytest.mark.parametrize("case", [(t1, d1, r1), (t2, d2, r2), (t3, d3, r3),
+                                      (t4, d4, r4), (t5, d5, r5)])
+    def test_right_censored_against_examples(self, case):
+        times, died, ref = case
+        sample = stats.CensoredData.right_censored(times, np.logical_not(died))
```

Comment:
I see this is what I missed, that you were inverting here with `np.logical_not`. All good then 👍